src/devices/machine/k056230.cpp

```diff
@@ -119,6 +152,316 @@ void k056230_device::ram_w(offs_t offset, u32 data, u32 mem_mask)
 	COMBINE_DATA(&lanc_ram[offset & 0x7ff]);
 }
 
+
+void k056230_device::set_mode(u8 data)
+{
+	switch (data & 0xf0)
+	{
+		case 0x00:
+		case 0x20:
+			m_linkid = data & 0x0f;
+			break;
+
+		default:
+			logerror("k056230-set_mode: %02x\n", data);
+			break;
+	}
+}
+
+void k056230_device::set_ctrl(u8 data)
+{
+	m_linkenable = data && 0x10;
```

Comment:
I’m pretty sure this isn’t what you want to do.  You’re testing that `data` is non-zero and the literal constant 0x10 is non-zero (which is isn’t).

Reply:
well... good catch - although it worked ;)
it more likely is meant to be `m_linkenable = (data & 0x10) ? 1 : 0;`